From 8cfd1a1db021fff3575dc835bcafea54b38a00cb Mon Sep 17 00:00:00 2001
From: Jonas Kvinge <jonas@jkvinge.net>
Date: Thu, 28 Dec 2023 22:38:24 +0100
Subject: [PATCH] pcre2-config.cmake.in: Fix PCRE2 library names

Fix library names for 16bit, 32bit and posix which incorrectly points to 8bit.
Also add IMPORTED_IMPLIB to fix https://bugreports.qt.io/browse/QTBUG-117907
---
 cmake/pcre2-config.cmake.in | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/cmake/pcre2-config.cmake.in b/cmake/pcre2-config.cmake.in
index b313d6df7..12f3a35ad 100644
--- a/cmake/pcre2-config.cmake.in
+++ b/cmake/pcre2-config.cmake.in
@@ -52,9 +52,9 @@ else ()
   endif ()
 endif ()
 find_library(PCRE2_8BIT_LIBRARY NAMES ${PCRE2_PREFIX}${PCRE2_8BIT_NAME}${PCRE2_SUFFIX} ${PCRE2_PREFIX}${PCRE2_8BIT_NAME}d${PCRE2_SUFFIX} DOC "8 bit PCRE2 library")
-find_library(PCRE2_16BIT_LIBRARY NAMES ${PCRE2_PREFIX}${PCRE2_16BIT_NAME}${PCRE2_SUFFIX} ${PCRE2_PREFIX}${PCRE2_8BIT_NAME}d${PCRE2_SUFFIX} DOC "16 bit PCRE2 library")
-find_library(PCRE2_32BIT_LIBRARY NAMES ${PCRE2_PREFIX}${PCRE2_32BIT_NAME}${PCRE2_SUFFIX} ${PCRE2_PREFIX}${PCRE2_8BIT_NAME}d${PCRE2_SUFFIX} DOC "32 bit PCRE2 library")
-find_library(PCRE2_POSIX_LIBRARY NAMES ${PCRE2_PREFIX}${PCRE2_POSIX_NAME}${PCRE2_SUFFIX} ${PCRE2_PREFIX}${PCRE2_8BIT_NAME}d${PCRE2_SUFFIX} DOC "8 bit POSIX PCRE2 library")
+find_library(PCRE2_16BIT_LIBRARY NAMES ${PCRE2_PREFIX}${PCRE2_16BIT_NAME}${PCRE2_SUFFIX} ${PCRE2_PREFIX}${PCRE2_16BIT_NAME}d${PCRE2_SUFFIX} DOC "16 bit PCRE2 library")
+find_library(PCRE2_32BIT_LIBRARY NAMES ${PCRE2_PREFIX}${PCRE2_32BIT_NAME}${PCRE2_SUFFIX} ${PCRE2_PREFIX}${PCRE2_32BIT_NAME}d${PCRE2_SUFFIX} DOC "32 bit PCRE2 library")
+find_library(PCRE2_POSIX_LIBRARY NAMES ${PCRE2_PREFIX}${PCRE2_POSIX_NAME}${PCRE2_SUFFIX} ${PCRE2_PREFIX}${PCRE2_POSIX_NAME}d${PCRE2_SUFFIX} DOC "8 bit POSIX PCRE2 library")
 unset(PCRE2_NON_STANDARD_LIB_PREFIX)
 unset(PCRE2_NON_STANDARD_LIB_SUFFIX)
 unset(PCRE2_8BIT_NAME)
@@ -126,6 +126,7 @@ if (PCRE2_FOUND)
     endif ()
     set_target_properties(PCRE2::${component} PROPERTIES
       IMPORTED_LOCATION "${PCRE2_${component}_LIBRARY}"
+      IMPORTED_IMPLIB "${PCRE2_${component}_LIBRARY}"
       INTERFACE_INCLUDE_DIRECTORIES "${PCRE2_INCLUDE_DIR}"
     )
     if (component STREQUAL "POSIX")
